tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,297 @@
+                                       rd_kafka_conf_dup(conf), NULL);
+        rd_kafka_conf_destroy(conf);
+
+        test_consumer_subscribe(c[0].rk, topics);
+        test_consumer_subscribe(c[1].rk, topics);
+
+        /*
+         * Static members continue to enforce `max.poll.interval.ms`.
+         * These members remain in the member list however so we must
+         * interleave calls to poll while awaiting our assignment to avoid
+         * unexpected rebalances being triggered.
+         */
+        rebalance_start = test_clock();
+        while (!static_member_wait_rebalance(&c[0], rebalance_start,
+              &c[0].assigned_at, 1000))
+                /* keep consumer 2 alive while consumer 1 awaits its assignment */
+                test_consumer_poll_once(c[1].rk, &mv, 0);
+
+        static_member_wait_rebalance(&c[1], rebalance_start,
+                                     &c[1].assigned_at, -1);
+
+        /*
+         * Consume all the messages so we can watch for duplicates
+         * after rejoin/rebalance operations.
+         */
+        test_consumer_poll("serve.queue", 
+                           c[0].rk, testid, c[0].partition_cnt, 0, -1, &mv);
+        test_consumer_poll("serve.queue", 
+                           c[1].rk, testid, c[1].partition_cnt, 0, -1, &mv);
+
+        test_msgver_verify("first.verify", &mv, TEST_MSGVER_ALL, 0, msgcnt);
+
+        TEST_SAY("== Testing consumer restart ==\n");
+        /* Only c[1] should exhibit rebalance behavior */
+        c[1].max_rebalance_cnt++;
+
+        conf = rd_kafka_conf_dup(rd_kafka_conf(c[1].rk));
+        test_consumer_close(c[1].rk);
+        rd_kafka_destroy(c[1].rk);
+
+        c[1].rk = test_create_handle(RD_KAFKA_CONSUMER, conf);
+        rd_kafka_poll_set_consumer(c[1].rk);
+        test_consumer_subscribe(c[1].rk, topics);
+
+        /* Await assignment */
+        rebalance_start = test_clock();
+        while (!static_member_wait_rebalance(&c[1], rebalance_start,
+               &c[1].assigned_at, 1000))
+                test_consumer_poll_once(c[0].rk, &mv, 0);
+
+        TEST_SAY("== Testing subscription expansion ==\n");
+        /* 
+         * New topics matching the subscription pattern should cause
+         * group rebalance
+         */
+        c[0].max_rebalance_cnt++;
+        c[1].max_rebalance_cnt++;
+
+        rebalance_start = test_clock();
+        test_create_topic(c->rk, tsprintf("%snew", topic), 1, 1);
+
+        /* Await revocation */
+        while (!static_member_wait_rebalance(&c[0], rebalance_start,
+               &c[0].revoked_at, 1000))
+                test_consumer_poll_once(c[1].rk, &mv, 0);
+
+        static_member_wait_rebalance(&c[1], rebalance_start,
+                                     &c[1].revoked_at, -1);
+
+        /* Await assignment */
+        while (!static_member_wait_rebalance(&c[0], rebalance_start,
+                                             &c[0].assigned_at, 1000))
```

Comment:
this indent is better, but also inconsistent with the previous identical while-loops.

Reply:
yeah I see I was aligning with the wrong opening paren